src/Eloquent/Model.php

```diff
@@ -250,9 +237,16 @@ public function setAttribute($key, $value)
     {
         $key = (string) $key;
 
-        // Add casts
-        if ($this->hasCast($key)) {
-            $value = $this->castAttribute($key, $value);
+        $casts = $this->getCasts();
+        if (array_key_exists($key, $casts)) {
+            $castType = $this->getCastType($key);
+            $castOptions = Str::after($casts[$key], ':');
+
+            // Can add more native mongo type casts here.
+            $value = match (true) {
+                $castType === 'decimal' => $this->fromDecimal($value, $castOptions),
+                default                 => $value,
+            };
```

Comment:
Thanks for the comment, what other [native types](https://www.mongodb.com/docs/manual/reference/bson-types/) do you think of?

If the goal is to have a map of `$castType` to native converter, the `match` may be rewrote like this:
```suggestion
            // Can add more native mongo type casts here.
            $value = match ($castType) {
                'decimal' => $this->fromDecimal($value, $castOptions),
                default   => $value,
            };
```